src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -17,6 +17,19 @@ const useFetchChannelHook = () => {
   const {localDb, refresh} = useLocalDatabaseHook();
   const {signedProfileId, anonProfileId} = useUserAuthHook();
 
+  const helperFollowingSystemMessage = (message) => {
+    if (message?.system_user === signedProfileId) {
+      message.text = message?.own_text;
+    } else {
+      message.text = message?.other_text;
+    }
+    /**
+     * TODO: Handle follow anonymous user system message later, if any
+     * */
+
+    return message;
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `helperFollowingSystemMessage` function directly modifies the input object `message`. This can lead to unexpected side effects if the original message object is used elsewhere in the code. It's generally a better practice to create a new object with the modified properties instead of mutating the input object directly.
```diff
-   const helperFollowingSystemMessage = (message) => {
-     if (message?.system_user === signedProfileId) {
-       message.text = message?.own_text;
-     } else {
-       message.text = message?.other_text;
-     }
-     return message;
-   };
+   const helperFollowingSystemMessage = (message) => {
+     let newText = message?.system_user === signedProfileId ? message?.own_text : message?.other_text;
+     return {...message, text: newText};
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->